lib/srv/db/dynamodb/engine.go

```diff
@@ -0,0 +1,421 @@
+/*
+
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package dynamodb
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/dax"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodbstreams"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	apiaws "github.com/gravitational/teleport/api/utils/aws"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/srv/db/common/role"
+	"github.com/gravitational/teleport/lib/utils"
+	libaws "github.com/gravitational/teleport/lib/utils/aws"
+)
+
+func init() {
+	common.RegisterEngine(newEngine, defaults.ProtocolDynamoDB)
+}
+
+// newEngine create new DynamoDB engine.
+func newEngine(ec common.EngineConfig) common.Engine {
+	return &Engine{
+		EngineConfig:  ec,
+		RoundTrippers: make(map[string]http.RoundTripper),
+	}
+}
```

Comment:
Nit: I think that we recently updated the logic how db engines are registered. 
https://github.com/gravitational/teleport/blob/master/lib/srv/db/server.go#L59

Reply:
will fix when I merge master in